pkg/controller/direct/bigquery/table_controller.go

```diff
@@ -0,0 +1,351 @@
+		log.V(2).Info("Removing schema from table definition because big query does not support setting schema on view creation")
+		schemaBack := table.Schema
+		table.Schema = nil
+		log.V(2).Info("Creating BigQuery without schema", "name", table.TableReference.TableId)
+		res, err := a.gcpService.Insert(parent.ProjectID, parent.DatasetID, table).Do()
+		if err != nil {
+			return fmt.Errorf("error creating Table %s: %w", a.id.ID(), err)
+		}
+		log.V(2).Info("successfully created Table", "name", res.Id)
+		table.Schema = schemaBack
+		log.V(2).Info("Updating BigQuery Table back with schema", "name", res.Id)
+		res, err = a.gcpService.Update(parent.ProjectID, parent.DatasetID, res.Id, table).Do()
+		if err != nil {
+			return fmt.Errorf("error updating Table %s: %w", a.id.ID(), err)
+		}
+		log.V(2).Info("successfully updated Table with schema", "name", res.Id)
+		if err := a.UpdateStatusForCreate(ctx, createOp, res); err != nil {
+			return err
+		}
+	} else {
+		insertCall := a.gcpService.Insert(parent.ProjectID, parent.DatasetID, table)
+		created, err := insertCall.Do()
+		if err != nil {
+			return fmt.Errorf("error creating Table %s: %w", a.id.ID(), err)
+		}
+		log.V(2).Info("successfully created Table", "name", a.id.String())
+		if err := a.UpdateStatusForCreate(ctx, createOp, created); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *Adapter) UpdateStatusForCreate(ctx context.Context, createOp *directbase.CreateOperation, created *bigquery.Table) error {
+	mapCtx := &direct.MapContext{}
+	status := BigQueryTableStatus_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	external := a.id.String()
+	status.ExternalRef = &external
+	if err := createOp.UpdateStatus(ctx, status, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("updating Table", "name", a.id.String())
+	mapCtx := &direct.MapContext{}
+
+	if err := a.normalizeReferences(ctx); err != nil {
+		return fmt.Errorf("normalizing reference for update: %w", err)
+	}
+
+	desired := a.desired.DeepCopy()
+	table := BigQueryTableSpec_ToProto(mapCtx, &desired.Spec)
+
+	// For table created from external data configuration,
+	// If no schema was provided, the Update request would fail.
+	// Getting schema from actual object.
+	if table.ExternalDataConfiguration != nil && table.Schema == nil && table.ExternalDataConfiguration.Schema == nil {
+		table.Schema = a.actual.Schema
+		for _, field := range table.Schema.Fields {
+			setEmptyPolicyTagsInSchema(field)
+		}
+	}
+	eq, err := TableEq(a.actual, table)
+
+	if err != nil {
+		return err
+	}
+	// No diff detected.
+	if eq {
+		return nil
+	}
+
+	a.customTableLogic(table)
+	parent := a.id.Parent()
+
+	if mapCtx.Err() != nil {
+		return fmt.Errorf("error updating Table %s: %w", a.id.ID(), mapCtx.Err())
```

Comment:
This error check should be moved up to  line 227, right?

And the error message is kinda confusing, should it just be `return mapCtx.Err()`?